frontend/src/metabase/query_builder/components/view/ConvertQueryModal/ConvertQueryModal.tsx

```diff
@@ -56,7 +53,9 @@ const ConvertQueryModal = ({
     const newDatasetQuery = createDatasetQuery(query, question);
     const newQuestion = question.setDatasetQuery(newDatasetQuery);
 
-    onUpdateQuestion?.(newQuestion, { shouldUpdateUrl: true });
+    // We should rerun the question to ensure we get the proper `mode.ModeFooter` since
+    // `mode` relies on `qb.lastRunCard`
+    onUpdateQuestion?.(newQuestion, { run: true, shouldUpdateUrl: true });
```

Comment:
Without rerunning the question we'd still show the mode footer even if the question is already changed to `native` and not a `timeseries` question anymore.

This is because the `mode` selector was relying on the previous run card. So without rerunning the card, we still get `timeseries` mode even though the question has turned native.

![Screenshot 2023-09-26 at 1 23 12 PM](https://github.com/metabase/metabase/assets/1937582/5978b3fa-8321-40a9-ac2a-37ae095d53c1)